MAVLink over local Unix FIFO

rsaxvc opened the pull request.

rsaxvc commented:
Should I rebase this to 2.4-evo?

Consti10 commented:
From glancing at the code, there are a couple of things that could be improved (and should before merging).

1) SerialEndpoint is only for a serial connection (and already complex enough doing only that)
Hacking a fifo mode into that is not ideal - you need to create its own endpoint
(Maybe FiFoEndpoint) for that
this also makes documentation easier, what exactly is the purpose of this endpoint, and how it should be used

2) We then need an extra mavlink param for enabling / disabling the fifo endpoint (or do it via the hardware.config file, since it is something that might not require enabling / disabling via qopenhd due to the rarity of people using it.

rsaxvc commented:
Thanks! I'll take a look at doing that.

Zipdox2 commented:
I don't think using a pipe is the best idea. I think using a Unix domain socket is a better solution. Here are some advantages:

- Supports multiple clients on one socket
- Bidirectional (RX and TX on the same socket)
- Supports datagrams (preserve message boundaries)

rsaxvc commented:
@Zipdox2 My use-case is replacing Mavlink over a UART and moving what used to be there on to the AirPi.

Mavlink has its own STX framing, so I'm not sure maintaining packet boundaries out of band is helpful. I don't have the code with me to check if the parser would take input with message boundaries.

A FIFO pair seemed to me like a good, simple fit, as I don't need connection tracking or multiple clients.

Zipdox2 commented:
You didn't mention your use case in the PR, it makes more sense now.

rsaxvc commented:
Thanks @Zipdox2 , I've updated the commit message to explain what it's used for.

I've also updated a first pass of moving this into its own endpoint. There's no setting yet.

Consti10 commented:
Can you tell me when the work on this endpoint is done ? I'l merge it into main then
And add the option to enable it via hardware.config

rsaxvc commented:
Sounds good, see you then!
